### DISCUSSION TITLE
Edge browser suggestion for google drive feature

### DISCUSSION THREAD

**Discussion by JeromeDeLeon** [penge/my-notes#397] [Ideas]
Hi, I'm still using this note extension for my day-to-day usage but the only problem I have since switching to the edge browser is the google drive sync feature. Having to switch to different devices, I have to export and import them, but even without it, I'm still okay using this extension as this has been my go-to note solution because of how easy it is to take notes.

Have you considered having a different UX for the edge browser? Like using this [approach](https://stackoverflow.com/questions/70187797/chrome-extension-crashes-the-microsoft-edge-browser-while-calling-launchwebauthf)?

Or having a dedicated profile feature that allows users to sign in/out and use that as a token? This way, it would be consistent with all browsers.

**Comment by penge** [comment]
Hi @JeromeDeLeon,
thank you very much for using My Notes since the early days! :)
Regarding Edge support, or the _**approach**_ you mentioned, I wish I'd have good news, but it doesn't seem to work for the following reasons:
1) Web application as OAuth client, needs to have a valid **origin** and **redirect URI**. Using `chrome://extension` is not valid. To solve that, Google Cloud offers a special application type just for this case, called **"Chrome app"**, which is exactly what My Notes is using.
2) The workaround describes creating a pop-up and listening to URL changes (this requires tabs permission) to read the received token. When I have tried that, got the same result as when I have tried `launchWebAuthFlow`, a window with a message saying it does not comply with OAuth policy.
3) [MV3](https://developer.chrome.com/docs/extensions/mv3/intro/) chrome extensions [do not](https://stackoverflow.com/questions/70819999/chrome-extension-manifest-3-https-apis-google-com) allow to load external script, like `apis.google.com` and then [auth2](https://github.com/google/google-api-javascript-client/blob/master/docs/reference.md), to have that [Google Sign-In button](https://developers.google.com/identity/sign-in/web/build-button).
4) For a moment, I thought I could use Firebase (product of Google) to authenticate and then use token to make requests towards [Google Drive](https://stackoverflow.com/questions/68571889/how-to-use-firebase-idtoken-to-generate-oauth-access-for-google-drive), but [some steps](https://firebaseopensource.com/projects/firebase/quickstart-js/auth/chromextension/readme/) worked only up to MV2.

It all goes down to Chrome preferring Google account, and Edge preferring Microsoft account.
Even if `chrome.identity.getAuthToken` would be implemented in Edge, it would most likely be linked to Microsoft services, therefore not to Google Drive. The only solution seems to not use Google Drive, and to use different sign-in (and registrations) method that would work across browsers.

**Reply by JeromeDeLeon** [reply]
Thank you for putting up these reasons. I agree with all of these and I guess writing your own auth would require tons of work.
